src/WLR/Render/Texture.hsc

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE EmptyDataDeriving, PatternSynonyms #-}
+
+module WLR.Render.Texture where
+
+#define WLR_USE_UNSTABLE
+-- Include the interface header, which in turn includes the desired
+-- <wlr/render/wlr_texture.h>. This makes it possible to define
+-- WLR_texture_impl in this file, avoiding recursive imports
+#include <wlr/render/interface.h>
+
+import Foreign
+import Foreign.C.Types
+
+import PIXMAN.Pixman
+import WLR.Render.Renderer
+import WLR.Types.Buffer
+
+{{ struct
+    wlr/render/wlr_texture.h,
+    wlr_texture,
+    impl,     Ptr WLR_texture_impl,
+    width,    CUInt,
+    height,   CUInt,
+    renderer, Ptr WLR_renderer
+}}
+
+type FunUpdateFromBuffer
+    =  Ptr WLR_texture
+    -> Ptr WLR_buffer
+    -> Ptr PIXMAN_region32_t
+    -> IO (CBool)
+
+{{ struct
+    wlr/render/interface.h,
+    wlr_texture_impl,
+    update_from_buffer, FunPtr FunUpdateFromBuffer,
+    destroy,            FunPtr (Ptr WLR_texture -> IO ())
+}}
```

Comment:
This should go in `WLR.Render.Interface`, I think.

Reply:
It should, but if it does go there then we'll wind up with recursive imports. If you've got an idea for how to handle that, then by all means let me know.

Reply:
One of the other PRs ran into a situation with cyclic imports, and fixed it by adding a `.hs_boot` file. Not entirely sure how it works, but it looks quite simple to do.